app/user.go

```diff
@@ -257,11 +257,21 @@ func (c *UserClient) setAccountRole(
 		return err
 	}
 	var newRoleIDs []string
-	accountRoleToSet, err := getAccountRole(c.ctx, c.client, accountRole)
+	accountRoleToSet, err := getAccountRole(c.ctx, c.client, accountRole, true)
 	if err != nil {
 		return err
 	}
-	if accountRoleToSet.Spec.AccountRole.ActionGroup == auth.ACCOUNT_ACTION_GROUP_ADMIN {
+	if accountRoleToSet == nil {
```

Comment:
if possible, i'd recommend having `getAccountRole` return `auth.ACCOUNT_ACTION_GROUP_UNSPECIFIED` instead of `nil`.